gql/parser.go

```diff
@@ -562,20 +580,182 @@ func parseArguments(l *lex.Lexer) (result []pair, rerr error) {
 	return result, nil
 }
 
+// String converts FilterTree to a string. Good for testing, debugging.
+func (t *FilterTree) String() string {
```

Comment:
Is this for debugging purposes? Because you can avoid string conversion in production run.

If this is for debug purposes, then call it debugString(), and make it internal.